packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -97,6 +103,80 @@ export async function generateArmTemplate(ctx: SolutionContext): Promise<Result<
   return result;
 }
 
+type DeployContext = {
+  ctx: SolutionContext;
+  finished: boolean;
+  client: ResourceManagementClient;
+  resourceGroupName: string;
+  deploymentStartTime: number;
+  deploymentName: string;
+};
+
+async function pollDeploymentStatus(deployCtx: DeployContext) {
+  while (!deployCtx.finished) {
+    const deployments = await deployCtx.client.deployments.listByResourceGroup(
+      deployCtx.resourceGroupName
+    );
+    const status = extractDeploymentStatus(deployCtx, deployments);
+    deployCtx.ctx.logProvider?.info(
+      format(
+        getStrings().solution.DeployArmTemplates.PollDeploymentStatusNotice,
+        PluginDisplayName.Solution
+      )
+    );
+    if (status.get(ArmDeploymentStatus.Running)?.length) {
+      deployCtx.ctx.logProvider?.info(
+        `[${PluginDisplayName.Solution}] Running deployments-> ${status
+          .get(ArmDeploymentStatus.Running)
+          ?.join(",")}`
+      );
+    }
+    if (status.get(ArmDeploymentStatus.Succeeded)?.length) {
+      deployCtx.ctx.logProvider?.info(
+        `[${PluginDisplayName.Solution}] Succedded deployments-> ${status
+          .get(ArmDeploymentStatus.Succeeded)
+          ?.join(",")}`
+      );
+    }
+    if (status.get(ArmDeploymentStatus.Failed)?.length) {
```

Comment:
Can we only do special treatment to failure situation and leave other status have some behavior? The code seems redundant here.

Reply:
It is fixed. Thanks.